vector/src/main/java/im/vector/app/features/link/LinkHandlerActivity.kt

```diff
@@ -54,20 +59,59 @@ class LinkHandlerActivity : VectorBaseActivity() {
             return
         }
 
-        if (uri.path == PATH_CONFIG) {
-            if (sessionHolder.hasActiveSession()) {
-                displayAlreadyLoginPopup(uri)
-            } else {
-                // user is not yet logged in, this is the nominal case
-                startLoginActivity(uri)
-            }
+        if (uri.getQueryParameter(CONFIG_PATH_HS_PARAMETER) != null) {
+            handleConfigUrl(uri)
+        } else if (SUPPORTED_HOSTS.contains(uri.host)) {
+            handleSupportedHostUrl(uri)
+        }
+    }
+
+    private fun handleConfigUrl(uri: Uri) {
+        if (sessionHolder.hasActiveSession()) {
+            displayAlreadyLoginPopup(uri)
         } else {
-            // Other link are not yet handled, but should not comes here (manifest configuration error?)
-            Timber.w("Unable to handle this uir: $uri")
+            // user is not yet logged in, this is the nominal case
+            startLoginActivity(uri)
+        }
+    }
+
+    private fun handleSupportedHostUrl(uri: Uri) {
+        if (!sessionHolder.hasActiveSession()) {
+            startLoginActivity(uri)
             finish()
+        } else {
+            convertUriToPermalink(uri)?.let { permalink ->
+                startPermalinkHandler(permalink)
+            } ?: run {
+                // Host is correct but we do not recognize path
+                Timber.w("Unable to handle this uri: $uri")
+                finish()
+            }
         }
     }
 
+    private fun convertUriToPermalink(uri: Uri): String? {
+        val path = SUPPORTED_PATHS.find { it in uri.toString() } ?: return null
+        // https://riot.im/develop/#/room/#element-android:matrix.org ->  https://matrix.to/#/#element-android:matrix.org
+        // https://app.element.io/#/room/#element-android:matrix.org  ->  https://matrix.to/#/#element-android:matrix.org
+        return uri
+                .toString()
+                .replace(uri.toString().substring(0, uri.toString().indexOf(path) + path.length), "https://$MATRIX_TO_HOST/#")
```

Comment:
You can simply write
`return "https://$MATRIX_TO_HOST/#" + uri.toString().substringAfter(path)`